javalib/src/main/scala/java/lang/String.scala

```diff
@@ -633,18 +633,49 @@ final class _String()
     buffer
   }
 
-  def toLowerCase(locale: Locale): _String = ???
+  def toLowerCase(locale: Locale): _String = {
+    toCase(locale, Character.toLowerCase)
+  }
 
   def toLowerCase(): _String =
     toLowerCase(Locale.getDefault)
 
   override def toString(): String = this
 
-  def toUpperCase(locale: Locale): _String = ???
-
+  def toUpperCase(locale: Locale): _String = {
+    toCase(locale, Character.toUpperCase)
+  }
   def toUpperCase(): _String =
     toUpperCase(Locale.getDefault)
 
+  private[this] def toCase(locale: Locale, convert: Int => Int): _String = {
+    if (count == 0) value
```

Comment:
In the Harmony 6.0 there was a special case for Turkish. The newer Harmony https://github.com/apache/harmony/blob/java6/classlib/modules/luni/src/main/java/java/lang/String.java defers to icu4j `UCharacter` which defers to UCaseProps which has a couple of special cases for Turkish and Lithuanian but I think this should gets us pretty close. Overall, this stuff is really complex especially icu4j which I guess is a complete Unicode implementation.